ores/score_processors/celery.py

```diff
@@ -102,7 +112,21 @@ def _generate_id(self, context, model, rev_id):
 
         return ":".join(str(v) for v in [context, model, rev_id, version])
 
+    def _queue_full(self):
+        # Check redis to see if the queue is too big.  If not, lump it on!
+        # This will result in a race condition, but it should have OK
+        # properties
+        if self.redis is not None:
+            queue_size = self.redis.llen("celery")
+            logger.warning("Queue size is too full {0}".format(queue_size))
+            return queue_size > self.queue_maxsize
+        else:
+            return False
+
     def _score(self, context, model, rev_ids, caches=None):
+        if self._queue_full():
+            raise queue.Full()
```

Comment:
Is this just an exception? I wonder if we should have a specialized exception here for this instead, since queue.Full() probably has a specific meaning about the queue module etc than a generic queue system.... But if this makes sense to you I'm cool with it.